### PR TITLE
Milestone 3: Feature/events

### PR DESCRIPTION
**Pull request by SarvarKh**
# Magic Books Inc - Milestone 3: Feature/events
**In this milestone, we have:**
- [x] made the BooksForm component stateful;
- [x] connected BoosForm with redux store;
- [x] implemented ADD_BOOK and REMOVE_BOOK to the components;
- [x] redeployed to Heroku;
- [x] fixed linter errors;

### CODE REVIEW

File: src/components/Book.jsx [hunk 1]
```diff
@@ -1,16 +1,23 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
```
Comment (bellom) [inline src/components/Book.jsx]: Opps 🤦 , let's stick to best practice and project guidelines.

Let's remove this line, we don't need it so far we are passing the method down to the Book component as a prop.

In milestone 2, you were introduced to the `component` and `container` folders.

And it was stated that the `book.jsx` would be a presentational component.

Note:  Presentational components for stateless components and a container/page folder for other components that use the state/redux-store.

More Explanation: 
React bindings for Redux separate presentational components from container components. This approach can make your app easier to understand and allow you to more easily reuse components. In your components directory (presentational components) you have components that are not connected to the Redux store, and on your containers directory, you have components that are connected to the Redux store. Read More... https://www.pluralsight.com/guides/how-to-structure-redux-components-and-containers

File: src/containers/BooksForm.js [hunk 2]
```diff
@@ -10,20 +13,53 @@ const CATEGORY = [
...
+      ISBN: Math.floor(Math.random() * 100000001),
+      title: titleInput,
+      category: categorySelect,
+    });
+    setTitleInput('');
+    setCategorySelect('');
+  };
+
+  return (
+    <form>
```
Comment (bellom) [inline src/containers/BooksForm.js]: For the booksForm handleSubmit function, It has to be attached to the form's onSubmit event. This way you can add validation to the input fields by adding the required attribute, and write less code as it is also one of the best practices.
 
 Screenshot from pathwright:
 
image

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**Kindly check my comments under this review;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best**

**Comment by chrystalme** [comment]
### Update
- [x]  Remove connect from book to make it presentation only component
- [x] Handle-Submit from form's event 

Thanks for taking the time to review our project 😃

**Review by oloomoses — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_Slack: Oloo Moses_

**Pull request merged**